RevenueCatUI/Scripts/PaywallsBehaviour.cs

```diff
@@ -1,32 +1,180 @@
-        /// <param name="options">Options for presenting the paywall.</param>
-        /// <returns>A <see cref="PaywallResult"/> describing the outcome.</returns>
-        public async Task<PaywallResult> PresentPaywallIfNeeded(string requiredEntitlementIdentifier, PaywallOptions options = null)
+        /// <param name="entitlementIdentifier">Entitlement identifier to check before presenting</param>
+        public async void PresentPaywallIfNeeded(string entitlementIdentifier)
+        {
+            if (string.IsNullOrEmpty(entitlementIdentifier))
+            {
+                Debug.LogError("[RevenueCatUI] Entitlement identifier cannot be null or empty.");
+                HandleResult(PaywallResult.Error);
+                return;
+            }
+
+            if (isPresenting)
+            {
+                Debug.LogWarning("[RevenueCatUI] Paywall is already being presented.");
+                return;
+            }
+
+            isPresenting = true;
+
+            try
+            {
+                var options = CreateOptions();
+                var result = await PaywallsPresenter.PresentIfNeeded(entitlementIdentifier, options);
+                HandleResult(result);
+            }
+            catch (Exception e)
+            {
+                Debug.LogError($"[RevenueCatUI] Exception in PaywallsBehaviour: {e.Message}");
+                HandleResult(PaywallResult.Error);
+            }
+            finally
+            {
+                isPresenting = false;
+            }
+        }
+
+        private PaywallOptions CreateOptions()
         {
-            return await PaywallsPresenter.PresentIfNeeded(requiredEntitlementIdentifier, options);
+            return new PaywallOptions
+            {
+                OfferingIdentifier = string.IsNullOrEmpty(offeringIdentifier) ? null : offeringIdentifier,
+                DisplayCloseButton = displayCloseButton
+            };
+        }
+
+        private void HandleResult(PaywallResult result)
+        {
+            if (result == null)
+            {
+                Debug.LogError("[RevenueCatUI] Received null PaywallResult.");
+                OnError?.Invoke();
+                return;
+            }
+
+            switch (result.Result)
+            {
+                case PaywallResultType.Purchased:
+                    OnPurchased?.Invoke();
+                    break;
+                case PaywallResultType.Restored:
+                    OnRestored?.Invoke();
+                    break;
+                case PaywallResultType.Cancelled:
+                    OnCancelled?.Invoke();
+                    break;
+                case PaywallResultType.NotPresented:
+                    OnNotPresented?.Invoke();
+                    break;
+                case PaywallResultType.Error:
+                    OnError?.Invoke();
+                    break;
+            }
         }
```

Comment:
Maybe it's worth logging if the callback is not set? 
For example:

` OnError?.Invoke();` if `OnError` is null, this will be lost

Reply:
what do we do if it's not set?